aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -147,9 +147,6 @@ Tensor masked_softmax(
     return at::_nested_tensor_softmax_with_shape(attn_scores, query);
   }
   if (attn_mask && attn_mask->dtype() != at::kBool) {
-    TORCH_WARN(
```

Comment:
I think there should be a way to avoid triggering this warning without actually removing it from the codebase.

Reply:
Supposedly not so well, and there were particular complaints about Torchscript.  But if you can suggest a solution.  
That being said, even then, the warning is sorta incorrect, because we end up warning the user even if she passed in a bool, because we converted it for them.)

Reply:
For posterity I think this is ok because I believe the only non-transformers usecase for `masked_softmax` where this warning might be useful is MaskedTensor